Sources/AsyncDNSResolver/c-ares/DNSResolver_c-ares.swift

```diff
@@ -272,7 +274,9 @@ extension Ares {
 
         init<Parser: AresQueryReplyParser>(parser: Parser, _ continuation: CheckedContinuation<Parser.Reply, Error>) {
             self._handler = { status, buffer, length in
-                guard status == ARES_SUCCESS || status == ARES_ENODATA else {
+                // print("---> Query callback status: \(status), buffer: \(buffer != nil), length: \(length)")
+                guard status == Int32(Int32(ARES_SUCCESS.rawValue)) || status == Int32(Int32(ARES_ENODATA.rawValue)) else {
```

Comment:
You've got an extra conversion to `Int32` here.

Reply:
Although it seems like a lot of the changes relate to checking the `ares_status_t` type. 

I think a better approach would be to define an `AresStatus` Swift `enum`. That would allow us to remove all of the extra integer conversion code we've added and allow us to check against Swift enum cases.